### PR TITLE
ruby 3

### PR DESCRIPTION
**Pull request by lvonk**
- Upgrade to ruby 3.0
- Let ~~travis~~ github actions run with ruby 3.0

### CODE REVIEW

**Comment by lvonk** [comment]
Fixes https://github.com/zilverline/sequent/issues/251

Locally one spec (always the first) fails from the `spec/lib/sequent/core/persistors/active_record_persistor_spec.rb` spec. The first time it tries to drop the in the test created database and fails because it is accessed by another user. The 2nd and 3rd test of the file can drop the created database.
Can not figure out yet why, let's see if the problem is the same on travis.

**Comment by lvonk** [comment]
Spec failed due to https://github.com/rails/rails/issues/40029, applied the work around and will continue to upgrade to latest AR. We should not use AR 6.0.3 I guess.

File: .ruby-version [hunk 1]
```diff
@@ -1 +1 @@
-2.7.1
+3.0.0
```
Comment (bforma) [inline .ruby-version:1]: 🚀

**Review by bforma — APPROVED** [review]
Looks good! 💪

**Pull request merged**